labels-config-service-impl/src/main/java/org/hypertrace/label/config/service/LabelsConfigServiceImpl.java

```diff
@@ -142,9 +153,11 @@ public void getLabel(GetLabelRequest request, StreamObserver<GetLabelResponse> r
   public void getLabels(
       GetLabelsRequest request, StreamObserver<GetLabelsResponse> responseObserver) {
     RequestContext requestContext = RequestContext.CURRENT.get();
-    List<Label> labelList = configServiceCoordinator.getAllLabels(requestContext);
-    labelList.addAll(systemLabels);
-    responseObserver.onNext(GetLabelsResponse.newBuilder().addAllLabels(labelList).build());
+    List<Label> allLabels = new ArrayList<>();
+    allLabels.addAll(systemLabels);
```

Comment:
Does the responsibility of merging this lie with the store? At first glance, seems to me just an internal store optimization that we're not writing the common ones. Are we ensuring they don't get updated too - can't recall?

Reply:
They are being picked up from config. This based on conversation we had during labels discussion.